src/server/scripts/EasternKingdoms/Stratholme/boss_baroness_anastari.cpp

```diff
@@ -88,37 +97,73 @@ class boss_baroness_anastari : public CreatureScript
             if (!UpdateVictim())
                 return;
 
-            //BansheeWail
-            if (BansheeWail_Timer <= diff)
-            {
-                if (rand32() % 100 < 95)
-                    DoCastVictim(SPELL_BANSHEEWAIL);
-                //4 seconds until we should cast this again
-                BansheeWail_Timer = 4000;
-            } else BansheeWail_Timer -= diff;
-
-            //BansheeCurse
-            if (BansheeCurse_Timer <= diff)
-            {
-                if (rand32() % 100 < 75)
-                    DoCastVictim(SPELL_BANSHEECURSE);
-                //18 seconds until we should cast this again
-                BansheeCurse_Timer = 18000;
-            } else BansheeCurse_Timer -= diff;
-
-            //Silence
-            if (Silence_Timer <= diff)
+            _events.Update(diff);
+
+            while (uint32 eventId = _events.ExecuteEvent())
             {
-                if (rand32() % 100 < 80)
-                    DoCastVictim(SPELL_SILENCE);
-                //13 seconds until we should cast this again
-                Silence_Timer = 13000;
-            } else Silence_Timer -= diff;
+                if (invisible)
+                {
+                    if (!possessedTarget->HasAura(SPELL_POSSESSED))
+                        _events.ScheduleEvent(EVENT_INVISIBLE, Seconds(0));
+
+                    if (possessedTarget->GetHealthPct() <= 50)
+                        _events.ScheduleEvent(EVENT_INVISIBLE, Seconds(0));
+                }
+                
+                switch (eventId)
+                {
+                    case EVENT_SPELL_BANSHEEWAIL:
+                        DoCast(me->GetVictim(),SPELL_BANSHEEWAIL);
+                        _events.ScheduleEvent(EVENT_SPELL_BANSHEEWAIL, Seconds(4));
+                        break;
+                    case EVENT_SPELL_BANSHEECURSE:
+                        DoCast(me->GetVictim(), SPELL_BANSHEECURSE);
+                        _events.ScheduleEvent(EVENT_SPELL_BANSHEECURSE, Seconds(18));
+                        break;
+                    case EVENT_SPELL_SILENCE:
+                        DoCast(me->GetVictim(), SPELL_SILENCE);
+                        _events.ScheduleEvent(EVENT_SPELL_SILENCE, Seconds(13));
+                        break;
+                    case EVENT_SPELL_POSSESS:
+                        possessedTarget = SelectTarget(SELECT_TARGET_RANDOM, 1, 0, true, false);
+                        if (possessedTarget->IsAlive())
```

Comment:
missing NULL check on possessedTarget (by taking a look at this code it seems it could be NULL)

Reply:
It will never be NULL because when you select target random it has an offset of 1. If no target is selected then possessedtarget will never be alive

Reply:
just use if (possessedTarget && possessedTarget->IsAlive()) to make sure that you wont get a nullpointer here to avoid crashes.

Reply:
This will crash if people go in alone. So yes.

Reply:
I tested alone and it didn't crash but ok I will change it